### PR TITLE
refactor: adjust exchange overview table - desktop

### PR DESCRIPTION
**Pull request by patricio0312rev**
# [[exchanges] adjust overview table](https://app.clickup.com/t/86du01z83)

## Summary

- Exchange overview table view and all its children components have been refactored to match the new designs.
- `twMerge` has been installed to handle custom class names overwrite.
- `TableCell.ts` has been refactored entirely, removing the problem with custom class names being overwritten by styled component styles.
- Snapshots for tests that included these components have been updated.

image

## Screenshots

- 768px:
image

- 1024px:
image

- 1280px:
image

## Checklist

-   [ ] My changes look good in both light AND dark mode
-   [ ] The change is not hardcoded to a single network, but has multi-asset in mind
-   [ ] I checked my changes for obvious issues, debug statements and commented code
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

**Comment by shahin-hq** [comment]
Hey @patricio0312rev, I know it is not ready to review yet but there is a thing you could easily fix:

image

notice how the line between rows are not reaching to the ends, you can fix easily by adding `with-x-padding` class to the parent of the `table` and `relative` class to `tr`s

https://github.com/ArdentHQ/arkvault/pull/618/files#diff-77d4b4f04e431b43ffaa86e16043e4961c791f30f2cfd5576e44c0df3c3a52aaR63

File: package.json [hunk 1]
```diff
@@ -93,6 +93,7 @@
 		"slick-carousel": "^1.8.1",
 		"socks-proxy-agent": "^7.0.0",
 		"string-hash": "^1.1.3",
+		"tailwind-merge": "^2.5.2",
```
Comment (patricio0312rev) [inline package.json:96]: I was having trouble with the `css` and `tw` props not overwriting all the class names in the styled component for the Table cell. Decided to directly install `twMerge` and refactor this component as a simple react component with twMerge handling their overwritten class names.

File: src/styles/base.css [hunk 2]
```diff
@@ -61,9 +61,9 @@ main {
 }
 
 .with-x-padding > table > tbody > tr:not(:last-child) td:first-child::after {
-	@apply content-[''] w-6 block bg-theme-background h-px absolute left-0
+	@apply content-[''] w-6 block bg-theme-background h-full absolute left-0
```
Comment (patricio0312rev) [inline src/styles/base.css:64]: When using a tooltip on the first cell of a column, the white padding line was always being placed in the middle (might be due to the floating position of the tooltip), causing the padding not to appear. However, using the full height of the cell fixed this.

**Review by samharperpittam — CHANGES_REQUESTED** [review]
Minor point but the "Time Ago" column text should be semi-bold to match the exchange column.

Design
image

Implementation
image

**Review by samharperpittam — CHANGES_REQUESTED** [review]
Also I think the tables are too big in height.

Designs should 12px of padding above and below the value 
image

However it looks like the table in implementation has 24px of padding above and below due to the additional space identified with an arrow here:
image

File: src/app/components/Amount/AmountLabel.tsx [hunk 3]
```diff
@@ -33,15 +30,16 @@ interface AmountLabelProperties {
...
-	let hintClassName = "bg-theme-success-200 dark:bg-theme-success-600";
+export const AmountLabel: React.VFC<AmountLabelProperties> = ({ value, ticker, isCompact, isNegative, hint, size }) => {
+	let labelColor = "success-bg";
+	let hintClassName = "bg-theme-success-500 dark:bg-theme-success-700 text-white";
 
 	if (isNegative) {
-		labelColor = "danger";
-		hintClassName = "bg-theme-danger-100 dark:bg-theme-danger-400";
+		labelColor = "danger-bg";
+		hintClassName = "bg-[#A56D4C] dark:bg-[#AA6868] text-white";
```
Comment (ItsANameToo) [inline src/app/components/Amount/AmountLabel.tsx:42]: do these purposely differ from the color options in the palette?

File: src/app/components/Label/Label.styles.tsx [hunk 4]
```diff
@@ -22,11 +22,15 @@ const getColor = (color?: ColorType, variant?: string) => {
 
 	const colors = {
 		danger: () => tw`text-theme-danger-400 border-theme-danger-100 dark:border-theme-danger-400`,
+		"danger-bg": () =>
+			tw`text-[#A56D4C] bg-[#FBEED6] dark:border dark:text-[#F39B9B] dark:bg-transparent dark:border-[#AA6868]`,
```
Comment (ItsANameToo) [inline src/app/components/Label/Label.styles.tsx:26]: same question here, not a fan of hex codes for colors 🤔 if they are different in the design, is there any color in the palette close enough to use for it instead?

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -488,7 +488,7 @@ jobs:
             - name: Test
               uses: nick-invision/retry@v2
               env:
-                  COVERAGE_THRESHOLD_BRANCHES: 99.67
+                  COVERAGE_THRESHOLD_BRANCHES: 99.66
```
[the remark this answers is not in the document]
Comment (ItsANameToo) [reply .github/workflows/test.yml:491]: @goga-m see my answer here https://github.com/ArdentHQ/arkvault/pull/618#discussion_r1726607337

**Review by ItsANameToo — APPROVED** [review]

**Review by samharperpittam — APPROVED** [review]

**Comment by ItsANameToo** [comment]
more conflicts @patricio0312rev 😂

**Pull request merged**